src/providers/data_provider_be.c

```diff
@@ -306,28 +350,23 @@ static void be_check_online_done(struct tevent_req *req)
     }
 }
 
-static void check_if_online(struct be_ctx *be_ctx)
+static void check_if_online(struct be_ctx *be_ctx, int delay)
 {
-    errno_t ret;
-
-    be_run_unconditional_online_cb(be_ctx);
-
-    if (!be_is_offline(be_ctx)) {
-        DEBUG(SSSDBG_TRACE_INTERNAL,
-              "Backend is already online, nothing to do.\n");
-        return;
-    }
+    struct tevent_timer *time_event;
+    struct timeval schedule;
 
     /* Make sure nobody tries to go online while we are checking */
     be_ctx->offstat.went_offline = time(NULL);
```

Comment:
This is not related directly to the patch, but what does `went_offline` do?
grep'ing source code I can only find places where it is set, but can't find where it is read/checked.
So I wonder if it would make sense to remove it totally (or to restore functionality).

Reply:
Looking at git history, the use of went_offline was removed on purpose with 45eb92f4564c76d04535b5c8e85e769db3ea93a3 so I agree it's safe to remove went_offline

Reply:
in case it wasn't obvious: I'm not proposing to remove went_offline in this PR. Maybe it would also be safer to open a ticket to not rush the change.

Reply:
I don't mind to address this comment (and comment about dereferencing uninitialized reply) in separate ticket(s)

Reply:
I created https://pagure.io/SSSD/sssd/issue/3990 to address those two comments.